### ISSUE TITLE
CurrentBranchName() get wrong name when head detached and git's language is not english

### ISSUE THREAD

**Issue by black-desk** [jesseduffield/lazygit#1467]
**Describe the bug**
as title

**To Reproduce**
1 set system language to a non-english language, for example chinese.
2 when head is detached, try to create a new branch on head

then you will get something like this:
fatal: '分离）' 不是一个提交，不能基于它创建分支 'tmp'

**Expected behavior**
create a new branch :)

**Screenshots**
![1](https://user-images.githubusercontent.com/29726878/132436042-2a98bdcb-845c-48be-af20-6cce4864a84f.png)
![2](https://user-images.githubusercontent.com/29726878/132436110-17f13fef-322a-4d06-9ccb-c4fe02cfb32e.png)
basically means that "分离）" is not a branch name

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.04
 - Lazygit Version: build from f44ae68e

**Additional context**
this problem cause by this line of code
https://github.com/jesseduffield/lazygit/blob/f44ae68e99386d5351d7cdec05a3231aa01a2d84/pkg/commands/branches.go#L31

the `CurrentBranchNameRegex` which is `(?m)^\*.*?([^ ]*?)\)?$` doesn't work with chinese, might not work with japanese as well :)

**Comment by black-desk** [comment]
ping @Ryooooooga
